addon/components/inputs/when.js

```diff
@@ -106,6 +108,19 @@ export default AbstractInput.extend({
     return get(cellConfig, 'renderer.size') || 'small'
   },
 
+  @readOnly
+  @computed('cellConfig')
+  timezone (cellConfig) {
+    return getWithDefault(cellConfig, 'renderer.timezone', '')
+  },
+
+  @readOnly
+  @computed('storedDateTimeValue', 'timezone')
+  displayStoredDateTimeValue (storedDateTimeValue, timezone) {
```

Comment:
Hmm. I realize now the `storedDateTimeValue` is a artifact of the `when` renderer which you probably just copied. `when` renderer does suffer from the same problems I mentioned.

Reply:
Ah yes, after closer inspection, I now see why the `storedDateTimeValue` is needed in the `when` renderer. I'll go back to using `value` in `datetime` but not sure if we can get rid of `storedDateTimeValue` in the `when` renderer

Reply:
Thanks. Stick with datetime for now. We'll address `when` at a future date.